### PR TITLE
ChunkedArray refactor

### PR DESCRIPTION
**Pull request by jwittenbach**
Refactored `ChunkedArray` to allow it to back Thunder's `Blocks` object.

The key change is that  `ChunkedArray.swap()` has been removed; replacing it are two more general functions: `keystovalues` and `valuestokeys`. `BoltArraySpark.swap()` now involves the following steps:
1. Create a `ChunkedArray` from the `BoltArraySpark` to break the values into chunks with `_chunk`
2. Use `keystovalues` to move some of the keys to the values
3. Use `valuestokeys` to move some of the values to the keys
4. Return put the values back together again with `unchunk`

During the `keystovalues` operation, we can specify if we would like to the new values to be chunked as well. If we choose not to chunk them and stop after step 2, then the resulting `ChunkedArray` is precisely what we want to back Thunder's `Blocks` object. This will be the pattern for both `Images.toSeries` and `Series.toImages`.

Then, depending on how we subsequently call `valuestokeys` we can implement `Blocks.toImages` or `Blocks.toSeries`.

### CODE REVIEW

File: bolt/spark/chunk.py [hunk 1]
```diff
@@ -136,80 +147,124 @@ def _unchunk(v):
...
+            rdd = rdd.mapValues(lambda v: squeeze(v))
+            newshape = self.shape[:-1]
+        else:
+            newshape = self.shape
+
+        return BoltArraySpark(rdd, shape=newshape, split=self._split)
 
-        Along with chunking, this is required as part of the swap method
-        on the BoltArraySpark, see there for further details.
+    def keystovalues(self, axes, size=None):
```
Comment (freeman-lab) [inline bolt/spark/chunk.py]: thoughts on renaming these `keys_to_values` and `values_to_keys`? i think i slightly prefer that.

Comment (jwittenbach) [inline bolt/spark/chunk.py]: That works for me. I'll to a little find-replace and push the changes.

**Comment by freeman-lab** [comment]
@jwittenbach I'm going to merge this in while I work on the `thunder` integration, I'll fix the Py3 errors in a follow-up.

**Comment by freeman-lab** [comment]
great work!

**Comment by jwittenbach** [comment]
@freeman-lab  Great, thanks! I did find one more edge-case last night that I added a test + fix for. Shall I just push it to `master/for-thunder`?

**Comment by freeman-lab** [comment]
@jwittenbach Sure sounds good! I just pushed the Python 3 fix.
